### PR TITLE
Add test for /admin/whois

### PR DESCRIPTION
**Pull request by illicitonion**
I accidentally pushed this to develop and reverted it.

Spec: https://github.com/matrix-org/matrix-doc/pull/198
Synapse: https://github.com/matrix-org/synapse/pull/418

### CODE REVIEW

File: tests/48admin.pl [hunk 1]
```diff
@@ -0,0 +1,31 @@
+test "/whois",
+   requires => [ $main::API_CLIENTS[0] ],
+
+   do => sub {
+      my ( $http ) = @_;
```
Comment (leonerd) [inline tests/48admin.pl]: Any reason this isn't

    requires => [ local_user_fixture() ],
    do => sub {
       my ( $user ) = @_;

?

File: tests/48admin.pl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+         ( $user ) = @_;
+
+         do_request_json_for( $user,
+            method => "GET",
+            uri    => "/api/v1/admin/whois/".$user->user_id,
+         )
+      })->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( $body, "devices", "user_id" );
```
Comment (leonerd) [inline tests/48admin.pl]: For consistency with the other tests this wants to be

    assert_json_keys( $body, qw( devices user_id ));

File: tests/48admin.pl (hunk 1 above)
Comment (illicitonion) [inline tests/48admin.pl]: Added comment

File: tests/48admin.pl (hunk 2 above)
Comment (illicitonion) [inline tests/48admin.pl]: Done

**Comment by leonerd** [comment]
OK, that comment makes sense.

Code LGTM aside from jenkins failure, which I presume is from branch mismatch.

**Pull request merged**